components/Context/DdfInfo.js

```diff
@@ -24,9 +28,158 @@ export default class DdfInfo extends React.Component {
+    issues.filter(issue => issue.contractAddress === -1);
+
+    issues.forEach(issue => {
+      const category = issue.category;
+      if(!processedIssues[category]){
+        processedIssues[category] = {issues: [issue], filters:{}};
+        categories.push(category);
+      }
+      else{
+        processedIssues[category].issues.push(issue);
+      }
+      issue.labels.forEach(label => {
+        //if the user has toggled the filter off before then let it be
+        if(this.state.issues && this.state.issues[category] && this.state.isssues[category].filters[label] === false){
+          processedIssues[category].filters[label] = false;
+        }
+        else if(processedIssues[category].filters[label] === undefined){
+          //filters always start off as checked
+          processedIssues[category].filters[label] = true;
+        }
+      })
+    });
+
+    this.setState({issues: processedIssues, categories})
+ }
+
+ getContractAddress(issue, comments){
+  //TODO Change from owner to status' username once jose implements the end point
+  comments.filter(comment => comment.author_association === "OWNER");
+  let contractAddress = -1;
+  let match = comments[0].body.match(ethereumRegex());
+  if(match && match.length > 0){
+    contractAddress = match[0];
+  }
+  return {
+    ...issue,
+    contractAddress,
+    value: "$100"
+  };
+ }
+
+async getCommentsOfIssue(repoName, number){
+  return new Promise(async (resolve, reject) => {
+    axios.get(`https://api.github.com/repos/csmartinsfct/${repoName}/issues/${number}/comments?access_token=573dbede2e9766fd4d41610d15a8b82371fd522a`)
+      .then(function (response) {
+        // handle success
+       resolve(response.data);
+      })
+      .catch(function (error) {
+        // handle error
+        console.log(error);
+      })
+    })
+ }
+
+ getCategory(repoName){
+  if(repoName.indexOf("tech") !== -1 || repoName.indexOf("website") !== -1)
+    return "Development";
+  else if(repoName.indexOf("design") !== -1){
+    return "Design";
+  }
+  else if(repoName.indexOf("marketing") !== -1)
+    return "Marketing";
+  //TODO Remove after implementing jose's end point - just because my repos don't have these new names yet :x
+  else if(repoName === "MyBit-Go")
+    return "Development";
+  else
+    return "Design";
+}
+
+ async getIssues(){
+  try{
+    let issuesTmp = await Promise.all(Repos.map(async (repoName) =>
+      axios.get(`https://api.github.com/repos/csmartinsfct/${repoName}/issues?access_token=573dbede2e9766fd4d41610d15a8b82371fd522a`)
+        .then(response => {
+          const data = response.data;
+          return data.map(issue => {
+            let indexOf = issue.html_url.indexOf('/issues');
+            const repoUrl = issue.html_url.substring(0, indexOf + 1);
+            const labels = issue.labels.map(label => label.name);
+            const repoName = issue.html_url.split('/')[4];
+            const category = this.getCategory(repoName);
```

Comment:
You just made your life harder, our API will not look like this, it's going to be way easier to consume. I mentioned many times we were not going to talk to GitHub directly...

Reply:
If you're just going to return github data and act like a proxy then its the same code to process either the request to your API or GitHub's?

Reply:
And if it won't be the same data then I can't write tests or anything either way because I don't know what the data will look like. I'm just going by what you told me the API would do - work like a proxy.